MCCoreDataStack/Library/CoreDataStackManager/MCCoreDataStackManager.swift

```diff
@@ -70,112 +70,93 @@ public struct StackManagerHelper {
+        self.domain = domain
         self.model = model
-        
-        if self.name.isEmpty { return nil }
     }
     
-    ///### Init method
-    ///- Parameter domainName: completion Block
-    ///- Parameter model: URL for the current data model
+    /// Init method
+    /// - Parameter domain: name of the current domain
+    /// - Parameter url: URL for the current data model
 
-    @objc public convenience init?(domainName: String, model URL: Foundation.URL?)
-    {
-        guard let model = NSManagedObjectModel.init(contentsOf: URL!) else {
-            
-            fatalError("Error initializing mom from: \(URL)")
+    @objc public convenience init?(domain: String, url: URL?) {
+        guard
+            let url = url,
+            let model = NSManagedObjectModel(contentsOf: url) else {
+            return nil
         }
         
-        self.init(domain: domainName, model: model)
+        self.init(domain: domain, model: model)
     }
-    
-    ///### This method delete the current persistent Store
-    ///- Parameter completionBlock: completion Block
 
-    @objc open func deleteStore(completionBlock: (() -> Void)?)
-    {
-        let store = self.PSC?.persistentStore(for: self.storeURL!)
-        if let storeUnwrapped = store {
-            do {
-                try self.PSC?.remove(storeUnwrapped)
-            } catch {}
-            
-            if FileManager.default.fileExists(atPath: (self.storeURL?.path)!) {
-                do {
-                    try FileManager.default.removeItem(atPath: (self.storeURL?.path)!)
-                    
-                    if let completionUnWrapped = completionBlock {
-                        completionUnWrapped();
-                    }
-                } catch { }
-            }
-        }
-    }
-    
-    //MARK: Private
-    
-    fileprivate func isPersistentStoreAvailable(_ completionBlock:MCCoreDataAsyncCompletion?)
-    {
-        if self.isReady {
-            if let completionBlockUnwrapped = completionBlock {
-                completionBlockUnwrapped();
-            }
-        } else {
-            let timeout: DispatchTime = DispatchTime.now() + Double(Int64(0.1 * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
+    // MARK: - Public methods
 
-            let _ = self.accessSemaphore.wait(timeout: timeout)
-            //FIXME: PersistentStore needs
-            let delayTime = DispatchTime.now() + Double(Int64(0.2 * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
-            DispatchQueue.main.asyncAfter(deadline: delayTime) {
-                if let completionBlockUnwrapped = completionBlock {
-                    completionBlockUnwrapped();
-                }
-            }
+    /// Configure the current CoreDataStack
+    /// - Parameter url: the URL of your sqlite file. See StackManagerHelper for help.
+    /// - Parameter configuration: configuration name
+    @objc open func configure(url: URL, configuration: String?, completion: MCCoreDataAsyncCompletion?) {
```

Comment:
maybe it is better to change `url` to `storeURL` . At least it is more descriptive. If it was a `String` it was named as `storeUrl`